js/src/hooks/useAutoCreateAdsMCAccounts.js

```diff
@@ -0,0 +1,202 @@
+	 * isCreatingAccountsRef - Indicates if the accounts are being created.
+	 * accountsCreatedRef - Indicates if the accounts have been created.
+	 */
+	const isCreatingBothAccountsRef = useRef( false );
+	const isCreatingAdsAccountsRef = useRef( false );
+	const isCreatingMCAccountsRef = useRef( false );
+	const initHasExistingMCAccountsRef = useRef( null );
+	const initHasExistingAdsAccountsRef = useRef( null );
+	const accountsCreatedRef = useRef( false );
+
+	const {
+		data: existingMCAccounts,
+		hasFinishedResolution: hasFinishedResolutionForExistingMCAccounts,
+	} = useExistingGoogleMCAccounts();
+
+	const {
+		existingAccounts: existingAdsAccount,
+		hasFinishedResolution: hasFinishedResolutionForExistingAdsAccount,
+	} = useExistingGoogleAdsAccounts();
+
+	const {
+		hasFinishedResolution: hasFinishedResolutionForGoogleAdsAccount,
+		hasGoogleAdsConnection,
+	} = useGoogleAdsAccount();
+
+	const {
+		googleMCAccount,
+		hasFinishedResolution: hasFinishedResolutionForGoogleMCAccount,
+	} = useGoogleMCAccount();
+
+	const [ handleCreateAccount, { response } ] = useCreateMCAccount();
+	const [ upsertAdsAccount, { loading } ] = useUpsertAdsAccount();
+	const isGoogleMCConnected =
+		googleMCAccount?.status === GOOGLE_MC_ACCOUNT_STATUS.CONNECTED ||
+		googleMCAccount?.status === GOOGLE_MC_ACCOUNT_STATUS.INCOMPLETE;
+
+	const hasExistingMCAccount =
+		isGoogleMCConnected || existingMCAccounts?.length > 0;
+	const hasExistingAdsAccount =
+		hasGoogleAdsConnection || existingAdsAccount?.length > 0;
+
+	if (
+		initHasExistingMCAccountsRef.current === null &&
+		hasFinishedResolutionForExistingMCAccounts &&
+		hasFinishedResolutionForGoogleMCAccount
+	) {
+		initHasExistingMCAccountsRef.current = hasExistingMCAccount;
+	}
+
+	if (
+		initHasExistingAdsAccountsRef.current === null &&
+		hasFinishedResolutionForExistingAdsAccount &&
+		hasFinishedResolutionForGoogleAdsAccount
+	) {
+		initHasExistingAdsAccountsRef.current = hasExistingAdsAccount;
+	}
+
+	const createMCAccount = useCallback( async () => {
+		await handleCreateAccount();
+	}, [ handleCreateAccount ] );
+
+	const createAdsAccount = useCallback( async () => {
+		await upsertAdsAccount();
+	}, [ upsertAdsAccount ] );
+
+	const createBothAccounts = useCallback( async () => {
+		await createMCAccount();
+		await createAdsAccount();
+	}, [ createMCAccount, createAdsAccount ] );
+
+	const accountCreationChecksResolved =
+		initHasExistingAdsAccountsRef.current !== null &&
+		initHasExistingMCAccountsRef.current !== null;
+
+	const shouldCreateAdsAccount =
+		initHasExistingAdsAccountsRef.current === false &&
+		initHasExistingMCAccountsRef.current === true;
+
+	const shouldCreateMCAccount =
+		initHasExistingAdsAccountsRef.current === true &&
+		initHasExistingMCAccountsRef.current === false;
+
+	const shouldCreateBothAccounts =
+		! initHasExistingAdsAccountsRef.current &&
+		! initHasExistingMCAccountsRef.current;
```

Comment:
I agree with your point as it might be confusing.

I have updated the property names as following.

1. `isCreatingBothAccounts`: Indicates if both the Google Ads and Google Merchant Center accounts are currently being created.
2. `isCreatingOnlyAdsAccount`: Indicates if only the Google Ads account is currently being created.
3. `isCreatingOnlyMCAccount`: Indicates if only the Google Merchant Center account is currently being created.

Also, updated the description for these properties in components. These names should be self-explanatory now.

Reply:
@eason9487 Now the hook is only returning two properties.

- `hasFinishedResolutionForExistingAdsMCAccounts` Indicates whether the checks for existing Merchant Center (MC) and Google Ads accounts have been completed.
- `isCreatingWhichAccount` The type of account that is being created. Possible values are 'ads', 'mc', or 'both'.